### PR TITLE
Cylinder and Torus primitives

### PR DESCRIPTION
**Pull request by lfrohman**
based on the code for the Sphere primitive.

### CODE REVIEW

File: rajawali/src/main/java/org/rajawali3d/primitives/Cylinder.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+				float x = mRadius * (float) Math.cos(verAngle);
+				float y = mRadius * (float) Math.sin(verAngle);
+
+				normals[vertIndex] = x * normLen;
+				vertices[vertIndex++] = x;
+				normals[vertIndex] = y * normLen;
+				vertices[vertIndex++] = y;
+				normals[vertIndex] = 0;
+				vertices[vertIndex++] = z;
+				System.out.println("normal["+((vertIndex-3)/3) +"]="+normals[vertIndex-3]+","+normals[vertIndex-2]+","+normals[vertIndex-1]);/////
```
Comment (jwoolston) [inline rajawali/src/main/java/org/rajawali3d/primitives/Cylinder.java]: System calls in general but especially System.out should be avoided on Android. In any event, this logging is verbose and should be removed/commented out.

File: rajawali/src/main/java/org/rajawali3d/primitives/Cylinder.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+public class Cylinder extends Object3D {
+
+	private final float PI = (float) Math.PI;
+	private float mLength;
+	private float mRadius;
+	private int mSegmentsC;
+	private int mSegmentsL;
+	private boolean mCreateTextureCoords;
+	private boolean mCreateVertexColorBuffer;
+	private Sphere s;
```
Comment (jwoolston) [inline rajawali/src/main/java/org/rajawali3d/primitives/Cylinder.java]: What is the purpose of this member?

**Comment by jwoolston** [comment]
All in all, a fantastic addition. Just a few clarifications/corrections and I will be happy to merge it.

**Comment by lfrohman** [comment]
Thanks, done,
     (I am kind of new at this).

**Comment by jwoolston** [comment]
@lfrohman No worries, we've all been there, and I consider this (peer review) the best possible way to learn. Thanks for the contribution!

**Comment by MasDennis** [comment]
Thanks for the contribution @lfrohman :beers: Much appreciated!

**Pull request merged**